### ISSUE TITLE
Test request:  space tank

### ISSUE THREAD

**Issue by gsalaman** [jlefkoff/JumbotronControlApp#3]
I'm seeing a weird behavior whilst implementing the MQTT shutdown command.  I'm curious to see whether the "spawn/kill" routines from your listener program work with Jeff's "fishtank" program.

If you've got some cycles (and it's easy), can you give this a shot?

**Comment by jlefkoff** [comment]
I am happy to give it a try, are you just using the python exit() command?

**Comment by gsalaman** [comment]
Yeah.  Looks like any form of exit (either exit(), sys.exit() or os._exit()) needs to be passed up through the mqtt callback to the overlying code wrapper...I'm guessing because of the fact the libraries we're using have multiple tasks/processes that they spawn...and they need to get "killed" from our high level process.  Jeff's program only has the RGB display library going, so if a direct "kill" works there, then it's likely the MQTT spawns, and not the RGB display ones.

Note that I've got it working by registering a separate "shutdown" callback triggered by the receipt of an MQTT message, and then passing that info up to the top level processing, which can then do a graceful exit() after cleaning up what it needs to.

**Comment by gsalaman** [comment]
Note I've also tried via raising various exceptions...same issue.  I think they need to hit the top level.
